### PR TITLE
Upgrade to rails 5

### LINKED ISSUE #1054: Eventually get to rails 5
I think https://github.com/comfy/comfortable-mexican-sofa/issues/728 was one of the key blockers, and that may have just turned solved - it would be good to understand what else blocks us.

### PR DESCRIPTION
**Pull request by Br3nda**
Following this https://hashrocket.com/blog/posts/how-to-upgrade-to-rails-5  
closes #1054

### CODE REVIEW

**Comment by Br3nda** [comment]
stuck at an error
`Sprockets::FileNotFound: couldn't find file 'jquery.ui.all' with type 'application/javascript'
`

**Comment by pozorvlak** [comment]
Weird. Have you run `rake assets:precompile`? It's not getting that far in Travis because it's failing at the `rake db:migrate` stage :-(

**Comment by Br3nda** [comment]
disabling comfortable mexican sofa for now -- it refers `require`s jquery in an outdated manner

**Comment by Br3nda** [comment]
352 failures to go

**Comment by CloCkWeRX** [comment]
> 352 failures to go
:P GO GO GO

File: app/models/member.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class Member < ActiveRecord::Base
+class Member < ApplicationRecord
```
[the remark this answers is not in the document]
Comment (Br3nda) [reply app/models/member.rb]: might need to factor some of member model out as a service

**Comment by Br3nda** [comment]
307 failures to go

**Comment by Br3nda** [comment]
243 to go

**Comment by Br3nda** [comment]
I'm thinking it's easier to just kickchart gem than solve our d3 errors.

**Comment by Br3nda** [comment]
Rather than work out d3 on rails 5 errors, i pulled in the kick chart gem to do charts for us. So we have less code to maintain here.

image

it was then easy to change to pie charts, which seem to make more sense to me. I'm looking at the charts for the most commonly planted sun/shade location, not the count.

so, turned to bar charts, and pushed up beside the predictions (which i styled a bit).

image

**Comment by Br3nda** [comment]
actually pretty keen to add a bunch more charts in other places, using that gem.

**Comment by Br3nda** [comment]
68 failures to go

**Comment by CloCkWeRX** [comment]
I like the charts better this way, I think - look a bunch snappier

**Comment by CloCkWeRX** [comment]
Worth turning bf34feacbafc98242b31decaaa5f40c4eaa018ce into a seperate PR and landing that now?

**Comment by Br3nda** [comment]
Kick chart gem, and pie chart now on  #1514

**Comment by Br3nda** [comment]
Note: Comfy mexican sofa is still disabled.

**Comment by cesy** [comment]
I think we're getting close - are you up for sorting the conflicts so we can try and get it complete this week?

**Comment by Br3nda** [comment]
Only the member deletion specs are failing. Deleted member's plantings etc still show up on pages. This is changed behaviour from rails 4 and probably needs a bigger fix

**Comment by Br3nda** [comment]
IT'S DONE!

**Comment by Br3nda** [comment]
I had to add validation that you own the item you're adding a photo to. Hence I had to change lots of tests to create model records with the same owner.

**Comment by Br3nda** [comment]
I've deployed this branch to heroku to make it easier to test
https://growstuff-rails5.herokuapp.com/

It's not got flickr/twitter/facebook secret keys yet, but the rest should be working

**Comment by Br3nda** [comment]
There's some elasticsearch errors. I think best approach might be upgrading to newest stable elasticsearch version, and gem. Then we reindex on production.

**Comment by cesy** [comment]
Upgrading elasticsearch sounds good to me.

**Comment by Br3nda** [comment]
underway on #1694

**Comment by Br3nda** [comment]
Almost a year since i opened this PR -- super hoping i can get it merged before Jan 3

File: app/controllers/gardens_controller.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class GardensController < ApplicationController
   before_action :authenticate_member!, except: %i(index show)
-  after_action :expire_homepage, only: %i(create delete)
+  after_action :expire_homepage, only: %i(create destroy)
```
Comment (Br3nda) [inline app/controllers/gardens_controller.rb]: this is the cause of the bug - where plantings from deleted gardens appear on the homepage

**Comment by Br3nda** [comment]
Geocoding is failing on gem version 1.5.0, so i went back to 1.4.9

**Comment by Br3nda** [comment]
@milesgould @pozorvlak @cesy IT's GREEN

File: .rubocop.yml [hunk 3]
```diff
@@ -31,17 +26,15 @@ Style/PercentLiteralDelimiters:
 Layout/MultilineMethodCallIndentation:
   EnforcedStyle: indented
 
+Layout/AlignHash:
+  EnforcedColonStyle: table
+  EnforcedHashRocketStyle: table
```
Comment (Br3nda) [inline .rubocop.yml]: there was a mixture of styles in the code base, so I choose one i liked.

Comment (pozorvlak) [reply .rubocop.yml]: I think you've earned that :-)

**Comment by cesy** [comment]
I haven't done a detailed review, but I've skimmed through enough as it was being done and trust you enough to merge, since we need this live.

**Comment by Br3nda** [comment]
Merged! Now to test test test staging.

**Pull request merged**